chacha20poly1305/src/cipher.rs

```diff
@@ -0,0 +1,122 @@
+//! Core AEAD cipher implementation for (X)ChaCha20Poly1305.
+
+// TODO(tarcieri): make this reusable for (X)Salsa20Poly1305
```

Comment:
Arguably the best way to do this is making it generic to all stream ciphers: https://github.com/RustCrypto/traits/issues/45